### ISSUE TITLE
nginx bitwarden.conf missing

### ISSUE THREAD

**Issue by banaantje** [bitwarden/server#2775]
### Steps To Reproduce

I've been setting up a clean Bitwarden Unified server. This is what I did:

- install docker & docker-compose
- create settings.env file and add/change the relevant entries applicable to my environment
- create docker-compose.yml with the necessary information
- docker compose up -d

Result is that two docker containers are running: bitwarden & mariadb.

Unfortunately the website is not accessible. When I bash into the docker container and to a "nginx -t" i get the error message:
nginx: [emerg] open() "/etc/nginx/http.d/bitwarden.conf" failed (2: No such file or directory) in /etc/nginx/nginx.conf:37

When the docker containers are being created eventually the 'entrypoint.sh' will be executed. This will execute the /usr/local/bin/hbs binary. This binary fails when I try it manually:

# /usr/local/bin/hbs
Unhandled exception: System.ArgumentException: Item has already been added. Key in dictionary: 'https_proxy'  Key being added: 'https_proxy'
   at System.Collections.Hashtable.Insert(Object key, Object nvalue, Boolean add)
   at System.Collections.Hashtable.Add(Object key, Object value)
   at Handlebars.conf.Program.AddEnvironmentVariablesToModel(IDictionary`2 model)
   at Handlebars.conf.Program.GetHandlebarsModel(Config config, Template template)
   at Handlebars.conf.Program.<>c.<<Main>b__0_1>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Invocation.AnonymousCommandHandler.InvokeAsync(InvocationContext )
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass17_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass12_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass19_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__18_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__5_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass8_0.<<UseExceptionHandler>b__0>d.MoveNext()

This error is also shown in the 'docker logs container' command.

Did I miss a certain step in the installation process?
Thanks for your time.

### Expected Result

the hbs command will use the files in the /etc/hbs directory to, amongst others, generate the bitwarden.conf. But it fails

### Actual Result

# /usr/local/bin/hbs
Unhandled exception: System.ArgumentException: Item has already been added. Key in dictionary: 'https_proxy'  Key being added: 'https_proxy'
   at System.Collections.Hashtable.Insert(Object key, Object nvalue, Boolean add)
   at System.Collections.Hashtable.Add(Object key, Object value)
   at Handlebars.conf.Program.AddEnvironmentVariablesToModel(IDictionary`2 model)
   at Handlebars.conf.Program.GetHandlebarsModel(Config config, Template template)
   at Handlebars.conf.Program.<>c.<<Main>b__0_1>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Invocation.AnonymousCommandHandler.InvokeAsync(InvocationContext )
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass17_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass12_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass19_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__18_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__5_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass8_0.<<UseExceptionHandler>b__0>d.MoveNext()

### Githash Version

N/A

### Environment Details

- OS: Debian

### Database Image

mariadb:10

### Issue-Link

https://github.com/bitwarden/server/issues/2480

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by jchristensen-bw** [comment]
Hi there,

I am unable to reproduce this issue, it has been escalated for further investigation.

Just to confirm, did you use the example files located at the following link, as templates when creating your settings.env, and docker-compose.yml files?
https://github.com/bitwarden/server/tree/master/docker-unified

Could you provide copies of your settings.env, and docker-compose.yml files with any sensitive values redacted?
This will help to further isolate the issue.

Additionally, in the bitwarden container, where you see the nginx error, can you manually check if the file referenced in the error exists?
/etc/nginx/http.d/bitwarden.conf

If you have more information that can help us, please add it below. 

Thanks!

**Comment by jchristensen-bw** [comment]
Thank you for providing those.

Looking at the settings.env file you provided, it looks like you aren't enabling any of the services:

```
# Services
# Some services, namely for enterprise use cases, are disabled by default. Defaults shown below.
#BW_ENABLE_ADMIN=true
#BW_ENABLE_API=true
#BW_ENABLE_EVENTS=false
#BW_ENABLE_ICONS=true
#BW_ENABLE_IDENTITY=true
#BW_ENABLE_NOTIFICATIONS=true
#BW_ENABLE_SCIM=false
#BW_ENABLE_SSO=false
```

The defaults are the ones set to true. At a minimum you'll want to enable the following:
BW_ENABLE_ADMIN
BW_ENABLE_API
BW_ENABLE_IDENTITY

**Comment by sso-bitwarden** [comment]
Hi @banaantje ,

- Is the container able to connect to the Internet and download files from github? 
- If you enter the unified container, are you able to run "/usr/local/bin/hbs"? Any error?

**Comment by joshtrichards** [comment]
@banaantje I noticed you overrode the webserver ports in your `settings.env` but did not change anything in the default `docker-compose.yml` that would suggest that was necessary.

Either re-comment out `BW_PORT_HTTP` and `BW_PORT_HTTPS` in your `settings.env` or explicitly set them to 8080+8443 (same difference) so they're back to their defaults. I'm fairly certain they're supposed to match the container ports rather than the host ports mapped in your `docker-compose.yml`.

If nothing else, those being incorrect are going keep anything from working.

P.S. The documentation on these settings could probably be expanded as I can see how changing them would appear to make sense.

**Comment by banaantje** [comment]
> Hi @banaantje ,
> 
>     * Is the container able to connect to the Internet and download files from github?
> 
>     * If you enter the unified container, are you able to run "/usr/local/bin/hbs"? Any error

**Comment by banaantje** [comment]
Helo All,

It's fixed. 

So what happend? At a certain point I got an error when I ran the hbs command that got me thinking.

Background story: previously we tested the regular bitwarden installation. Our server sits behind a squid proxy so we have to make some changes to the docker configuration to allow access through the proxy. One of those things we did was create a .docker/config.json file where we specify the http_proxy and https_proxy. This was a fix for the regular bitwarden installation.

I previously ran the hbs command and the error showed http_proxy. This was also seen in the docker logs. I did not made the link between the http_proxy error and the .docker/config.json file. But thanks to the comment of @joshtrichards and @sso-bitwarden I digged a bit further what was happening. That finally made the link. When I removed the config.json file, removed all the old bitwarden unified containers, and executed a "fresh" docker compose up -d everything started working.

Thanks for all your help! And sorry for the inconvenience caused.
